cmd/dep/ensure.go

```diff
@@ -304,27 +304,22 @@ func getProjectConstraint(arg string, sm gps.SourceManager) (gps.ProjectConstrai
 }
 
 func checkErrors(m map[string]pkgtree.PackageOrErr) error {
-	noGoErrors, pkgErrors := 0, 0
-	for _, poe := range m {
+	buildErrors, noGoErrors := []string{}, []string{}
```

Comment:
What do you think about using the following instead?

```go
var buildErrors []string
var noGoErrors []string
```

That way, if there are no errors, no memory is allocated for the two slices.